.github/workflows/ci.yml

```diff
@@ -99,20 +153,18 @@ jobs:
 
       - uses: actions/setup-python@v3
         with:
-          python-version: '3.8'
+          python-version: ${{ matrix.python }}
           architecture: 'x64'
       - uses: actions/cache@v3
         with:
           path: |
             ~/.cache/pip
             .nox
-          key: ${{ runner.os }}-2.5-${{ hashFiles('pyproject.toml') }}-${{ hashFiles('src/astro_databricks/__init__.py') }}
+          key: example-dags-os-${{ runner.os }}-python-${{ matrix.python }}-airflow-${{ matrix.airflow }}-deps-${{ hashFiles('pyproject.toml') }}-version-${{ hashFiles('src/astro_databricks/__init__.py') }}
       - run: cat .github/ci-test-connections.yaml > test-connections.yaml
       - run: sqlite3 /tmp/sqlite_default.db "VACUUM;"
       - run: pip3 install nox
-      - run: ls
-      - run: echo $PWD
-      - run: nox -s "test-3.8(airflow='2.5')" -- "tests/example_dags"
+      - run: DATABRICKS_GROUP_ID="${{github.run_id}}_${{matrix.python}}_${{matrix.airflow}}" nox -s "test-${{ matrix.python }}(airflow='${{ matrix.airflow }}')" -- "tests/test_example_dags.py"
```

Comment:
What do you think of also adding `--cov-report=xml`?

Could we also include:
```
      - name: Upload coverage
        uses: actions/upload-artifact@v2
        with:
          name: coverage-example-dags-${{ matrix.python }}-${{ matrix.airflow }}
          path: ./.coverage
```

Reply:
yes, added both suggesitons. thanks.